airflow/api_fastapi/serializers/connections.py

```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from pydantic import BaseModel, Field
+
+
+class ConnectionResponse(BaseModel):
+    """Connection serializer for responses."""
+
+    connection_id: str = Field(alias="conn_id")
+    conn_type: str
+    description: str | None
+    host: str | None
+    login: str | None
+    schema_: str | None = Field(alias="schema")
+    port: int | None
+    extra: str | None
```

Comment:
Extra need to go through the secret masker, in case some sensitive informations are contained in that field.  You can take a look at the original implementation of the marshmallow Schema for `Connection` to get an exemple of how to do that.

Reply:
@pierrejeambrun , I've added a field validator.

Reply:
Nice, maybe we can slightly update the test so we also have a redacted extra to be sure that secrets are not leaking.

This is pretty sensitive so I'd feel more confortable if this particular part was tested :)

Reply:
For some reason, redact method isn't redacting text. I'm using following test:

```
def test_get_should_respond_200_with_extra_redacted(self, test_client, session):
        self.create_connection()
        connection = session.query(Connection).first()
        connection.extra = '{"password": "test-password"}'
        session.commit()
        # breakpoint()
        response = test_client.get(f"/public/connections/{TEST_CONN_ID}")
        assert response.status_code == 200
        body = response.json()
        assert body["conn_id"] == TEST_CONN_ID
        assert body["conn_type"] == TEST_CONN_TYPE
        assert body["extra"] == '{"password": "****"}'
```

Reply:
Looks like this is because of `_disable_redact` in conftest.py. its set to autouse as a fixture. Looking into how to disable for a specific test

Reply:
fixed the test.